brave/brave-core
# Fix detection of Google Chrome Beta/Dev as default browsers (in non-English locales)

Resolves https://github.com/brave/brave-browser/issues/28717

Chrome Beta and Dev have translated default browser identifiers and they are living inside `google_chrome_strings.grd` as discussed with @mkarolin added code to `brave/script/chromium-rebase-l10n.py` to pull strings from `google_chrome_strings.grd` and adding them to `brave_strings.grd` with a modified IDS_, then pulling the translations from `google_chrome_strings_*.xtb` and adding them to `brave_strings_*.xtb` with a new matching  id attribute.

<img width="777" alt="image" src="https://user-images.githubusercontent.com/2965009/221936760-f66946ee-caba-4ac4-8948-b595a1c09dfd.png">


## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Set Google Chrome Beta or Dev as default browsers in the system with non-En locale
- Check brave://welcome and make sure they have default browser check mark
